jsonnet/drone.jsonnet

```diff
@@ -0,0 +1,68 @@
+// drone.jsonnet
+local drone = import 'github.com/Duologic/drone-libsonnet/main.libsonnet';
+
+local pipeline = drone.pipeline.docker;
+local step = drone.pipeline.docker.step;
+
+local image_to_push = 'grafana/wait-for-github';
+
+local images = {
+  alpine: 'alpine:3.17.0',
+  drone_plugin: 'plugins/docker',
+  drone_cli: 'drone/cli:latest',
+};
+
+local modified_paths = [
+  'go.mod',
+  'go.sum',
+  '**/*.go',
+];
+
+local pipelines = {
+  ['build_pipeline_%s' % (if main then 'main' else 'pr')]:
+    pipeline.new('build %spipeline' % (if main then 'and push ' else ''))
+    + pipeline.withSteps([
+      step.new('build + test%s' % (if main then ' + push' else ''), image=images.drone_plugin)
+      + step.withSettings({
+        dry_run: !main,
+        password: {
+          from_secret: 'docker-hub-password',
+        },
+        repo: image_to_push,
+        tags: 'latest',
+        username: {
+          from_secret: 'docker-hub-username',
+        },
+      }),
+    ])
+    + pipeline.trigger.onModifiedPaths(modified_paths)
+    + (if main then pipeline.trigger.onPushToMainBranch() else pipeline.trigger.onPullRequest())
+  for main in [true, false]
```

Comment:
So many ifs in there for only two cases, IMO a little repetition is not that bad. It would be more readable as two distinct pipelines 🤷 

WET (write everything twice) > DRY